beets/util/artresizer.py

```diff
@@ -232,23 +247,34 @@ def _check_method():
 
 def get_im_version():
     """Return Image Magick version or None if it is unavailable
-    Try invoking ImageMagick's "convert".
+    Try invoking ImageMagick's "magick". If "magick" is unavailable,
+    as with older versions, fall back to "convert"
+
+    Our iterator will be non-zero when the first command fails, and will
+    be returned in a tuple along with the version.
     """
-    try:
-        out = util.command_output(['convert', '--version'])
-
-        if b'imagemagick' in out.lower():
-            pattern = br".+ (\d+)\.(\d+)\.(\d+).*"
-            match = re.search(pattern, out)
-            if match:
-                return (int(match.group(1)),
-                        int(match.group(2)),
-                        int(match.group(3)))
-            return (0,)
-
-    except (subprocess.CalledProcessError, OSError) as exc:
-        log.debug(u'ImageMagick check `convert --version` failed: {}', exc)
-        return None
+    cmd_names = (['magick'],
+                 ['convert'])
+    for i, cmd_name in enumerate(cmd_names):
+
+        try:
+            cmd = cmd_name + ['--version']
+            out = util.command_output(cmd)
+
+            if b'imagemagick' in out.lower():
+                pattern = br".+ (\d+)\.(\d+)\.(\d+).*"
+                match = re.search(pattern, out)
+                version = (int(match.group(1)),
+                           int(match.group(2)),
+                           int(match.group(3)))
+                legacy = bool(i)
+                if match:
+                    return (version, legacy)
```

Comment:
No parentheses are necessary here:

```suggestion
                    return version, legacy
```

Reply:
Also, in the old version of the code, we still return a version (which just happens to be zero). So we probably want to preserve that behavior:

```py
else:
    return (0,), legacy
```